tsconfig.base.json

```diff
@@ -1,6 +1,6 @@
 {
   "compilerOptions": {
-    "lib": ["ES2022"],
+    "lib": ["ES2022", "DOM"],
```

Comment:
Had to add "DOM" here to access `document` in `useIsomorphicLayoutEffect`.

Reply:
We can't have this in our codebase, as it makes it far too easy to add in a mistaken reference to the DOM in an otherwise runtime agnostic library.

Let's change this before merging